### DISCUSSION TITLE
How can I get the size of an edge function?

### DISCUSSION THREAD

**Discussion by cachho** [vercel/community#5982] [Help] [answered]
### Summary

I'm wondering how I can find out if my edge function meets the requirements for vercel, or at least see the size.
Preferably locally, without building the whole application.

So far what I do is try to deploy on vercel and then see if it fails during the build process or not.

![](https://preview.redd.it/how-to-check-edge-function-size-for-vercel-v0-b0vdxi8phhlc1.png?width=1196&format=png&auto=webp&s=bbd2d788da7337ea3bdb230480f33533a7a28485)

The build process doesn't tell me which edge function is too big.

So now I'm in a situation where I changed a bunch of stuff and the build fails due to the edge function size, but I don't know if it's one of the new things I added or due to a changed dependency or what and I'll have to switch them all to nodejs and then one by one go back to edge.

What makes it worse is that it's also undeterministic and sometimes it's solved by simply retrying.

Does anyone know a good trick?

### Steps to Reproduce

Just create a bunch of edge functions.

**Comment by swarnava** [comment, marked as answer]
Hi,

I think this OOM issue is fixed on latest version of Next.js 14.1.1. Can you try it and let us know how it goes?

Typically you can run `vercel build` locally which generates `.vercel` folder in your local machine where you can review these information,

**Comment by JWW87** [comment]
I have the same question. When I deploy my project to Vercel, I get an error message that the edge function /api/v1/me/[sessionId] is too large, but the report shows that not a single page/API route is over 1MB in size (see screenshot). Also when I try to build locally with `vercel build` there is nothing larger than 1MB in the output. How can you verify the size of an edge function before a deploy on Vercel?

![Screenshot 2024-04-25 at 18-45-49 Dashboard](https://github.com/vercel/community/assets/39670111/874d0918-6f1d-45da-a88b-f0f042f407b0)
